govcd/catalog.go

```diff
@@ -219,7 +219,14 @@ func (cat *Catalog) UploadOvf(ovaFileName, itemName, description string, uploadP
 	uploadError := *new(error)
 
 	//sending upload process to background, this allows no to lock and return task to client
-	go uploadFiles(cat.client, vappTemplate, &ovfFileDesc, tmpDir, filesAbsPaths, uploadPieceSize, progressCallBack, &uploadError, isOvf)
+	go func() {
+		err = uploadFiles(cat.client, vappTemplate, &ovfFileDesc, tmpDir, filesAbsPaths, uploadPieceSize, progressCallBack, &uploadError, isOvf)
+		if err != nil {
+			util.Logger.Println(strings.Repeat("*", 80))
+			util.Logger.Printf("*** [DEBUG - UploadOvf] error calling uploadFiles: %s\n", err)
+			util.Logger.Println(strings.Repeat("*", 80))
+		}
+	}()
```

Comment:
Can we change/elaborate the old comment (few lines above) to reflect why we are doing this procedure this way?

Reply:
done